packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1824,13 +1829,40 @@ export class TransactionController extends BaseController<
       'updateTransactionGasFees',
     );
 
+    // Initialize gas fee values that will be used in the transaction update
+    let finalGasPrice = gasPrice;
+    let finalMaxFeePerGas = maxFeePerGas;
+    let finalMaxPriorityFeePerGas = maxPriorityFeePerGas;
+
+    const clonedTransactionMeta = cloneDeep(transactionMeta);
+    const isTransactionGasFeeEstimatesExists = transactionMeta.gasFeeEstimates;
+    const isAutomaticGasFeeUpdateEnabled =
+      this.#isAutomaticGasFeeUpdateEnabled(transactionMeta);
+    const shouldUpdateTxParamsGasFees =
+      isTransactionGasFeeEstimatesExists &&
+      isAutomaticGasFeeUpdateEnabled &&
+      userFeeLevel;
+
+    if (shouldUpdateTxParamsGasFees) {
+      applyTransactionGasEstimatesToTransaction({
+        txMeta: clonedTransactionMeta,
+        userFeeLevel: userFeeLevel as GasFeeEstimateLevel,
+      });
+
+      // Ignore the provided gas fee values when userFeeLevel is specified because
+      // the gas fee values are updated directly in transactionMeta in the applyTransactionGasEstimatesToTransaction
+      finalGasPrice = undefined;
+      finalMaxFeePerGas = undefined;
+      finalMaxPriorityFeePerGas = undefined;
+    }
+
     let transactionGasFees = {
       txParams: {
         gas,
         gasLimit,
-        gasPrice,
-        maxPriorityFeePerGas,
-        maxFeePerGas,
+        gasPrice: finalGasPrice,
```

Comment:
Minor, rather than using `finalX`, could we keep their actual property names to simplify this assignment?